src/client/options/mod.rs

```diff
@@ -1610,8 +1610,20 @@ impl ConnectionString {
             None => (None, None),
         };
 
-        let host_list: Result<Vec<_>> =
-            hosts_section.split(',').map(ServerAddress::parse).collect();
+        let host_list = hosts_section.split(',');
+        // checks if the usage of uds is correct
+        host_list.clone().try_for_each(|host| {
+            if host.ends_with(".sock") {
+                #[cfg(unix)]
+                percent_decode(host, "Unix domain sockets must be URL-encoded")?;
```

Comment:
`percent_decode` does not mutate its input, so we'll need to use a mutable iterator and reassign the value here.

Reply:
Thanks for your instructions!